server/src/server.ts

```diff
@@ -20,6 +22,14 @@ import { fileURLToPath } from "url";
 import { ChildProcess } from "child_process";
 import { WorkspaceEdit } from "vscode-languageserver";
 
+interface extensionConfiguration {
```

Comment:
curious: how does one decide whether this should be an interface or a type?

Reply:
It's pretty arbitrary, but they at least partly have different use cases. My understanding is interfaces have better perf and are recommended in general for object shapes.